draft/XXXX-orm/README.md

```diff
@@ -0,0 +1,103 @@
+---
+RFC: XXXX
+Author: Jacob Nesbitt
+Status: draft
+Created: 2020-07-28
+Last-Modified: 2020-07-28
+---
+
+# RFC XXXX: Multinet ORM
+
+This RFC addresses the issue of organizing database operations into classes which interact with each other. This aims to reduce the complexity, while improving the clarity and extendability of our database operations.
+
+## Motivation
+
+Currently, our database operations are structured in a purely functional way. That is, all database operations are done through simple functions, with no use of classes. This becomes problematic for readability, as our current `db.py` file has grown to almost 600 lines long. This presents a risk, as lack of code readability/clarity likely leads to more mistakes/bugs being introduced.
+
+Furthermore, due to the flat use of stateless functions, performance becomes an issue, as the same few functions are called repeatedly by multiple functions, as they share no state. The temporary solution to this was caching, but by introducing an ORM, much of the need for this caching will go away, as this state can be preserved in a class instance.
+
+## Proposals
+
+### Classes
+The following subsections describe the classes that make up the Multinet ORM. Type annotations are included to more accurately describe the connections between the classes. If a type annotation is left out, it is either due to uncertainty, or because that method returns nothing useful.
+
+
+##### User
+```
+# Previously used the term "cookie"
+def get_session() -> str: ...
+def set_session(): ...
+def delete_session(): ...
+
+# Serializes the object to JSON
+def serialize() -> Dict: ...
+
+# Static
+def exists(user: UserInfo) -> bool: ...
```

Comment:
Does this method test whether a user by the supplied `sub` value exists? How do you use an entire `UserInfo` to ask about whether that user exists?

I think we might also want a static method that performs a search by email address. What do you think?

Reply:
An example of where you'd use the entire `UserInfo` is when the user initially logs in through OAuth. We'll have the user information, but they wont exist in our database yet. If we were to have the sub value, we'd probably have the rest of the information as well.

I think a `search` method is a good idea, I can add it.

Reply:
Regarding search: 98ab5ff